Assets/Scripts/Player/PlayerCamera.cs

```diff
@@ -59,114 +82,160 @@ void Update()
+			                            cam.transform.right,
+			                            yMouseInput * _vertRotationSpeed * GetRotationSpeedModifier() );
+		}
 
-			Vector3 camEuler = cam.transform.eulerAngles;
-			camEuler.z = 0f;
+		KeepLineOfSight( actorHead, actorToCam );
 
-			if ( camEuler.x > rotBounds.min && camEuler.x < 300 )
-			{
-				camEuler.x = rotBounds.min;
-			}
-			if ( camEuler.x < rotBounds.max && camEuler.x > 100 )
-			{
-				camEuler.x = rotBounds.max;
-			}
+		// Lock Z rotation
+		Vector3 camEuler = cam.transform.eulerAngles;
+		camEuler.z = 0f;
+		cam.transform.eulerAngles = camEuler;
 
-			GetMinCameraDistance();
+		actor.SetRenderers( !IsCameraInsidePlayer() );
+	
+		Vector3 currentOffset = _focusOffset * _currentZoomDistance;
+		currentOffset += currentOffset * cam.nearClipPlane;
 
-			if ( actor.isRendererOn && currentCamDist < minCamDist )
-			{
-				actor.SetRenderers( false );
-			}
-			else if ( !actor.isRendererOn && currentCamDist >= minCamDist )
-			{
-				actor.SetRenderers( true );
-			}
+		cam.transform.position = Vector3.Lerp( cam.transform.position, 
+		                                       (_focusPoint - cam.transform.rotation * currentOffset) + Vector3.up * cam.nearClipPlane,
+		                                       _focusFollowSpeed * Time.deltaTime );
+	}
+
+	bool IsCameraInsidePlayer()
+	{
+		return _currentZoomDistance < _zoomDistanceBounds.min;
+	}
+
+	float GetRotationSpeedModifier()
+	{
+		float yInput = InputUtils.GetAxis( "Mouse Y" );
+		float xEuler = cam.transform.eulerAngles.x;
+		float speedMod = 1f;
 
-			Vector3 currentOffset = camOffset * currentCamDist;
-			targetPos = Vector3.Lerp( transform.position - cam.transform.rotation * currentOffset,
-			                          transform.position + Quaternion.Euler( -150.0f, camEuler.y, camEuler.z ) * currentOffset,
-			                          camEuler.x < 80.0f ? camEuler.x / 75.0f : 0.0f );
-
-			//RaycastHit hit = WadeUtils.RaycastAndGetInfo(cam.transform.position,
-			//                                             -cam.transform.up,
-			//                                             groundLayer,
-			//                                             groundCheckDist);
-			//if(hit.transform)
-			//{
-			//	targetPos.y = hit.point.y + groundCheckDist;
-			//}
-
-			cam.transform.position = targetPos + hitPosOffset;
-			cam.transform.eulerAngles = camEuler;
+		// Bottom rotation
+		if( xEuler > 270f && yInput < 0f ) 
+		{
+			xEuler -= 360f + _rotationBounds.min;
+			speedMod = Mathf.Lerp( 0f, 1f, Mathf.InverseLerp( 0f, _rotationBoundsWidth, xEuler ) );
+		}
+
+		// Top rotation
+		if( xEuler < 90f && yInput > 0f )
+		{
+			xEuler -= _rotationBounds.min;
+			speedMod = Mathf.Lerp( 1f, 0f, Mathf.InverseLerp( _rotationBounds.Difference - _rotationBoundsWidth,
+			                                              	  _rotationBounds.Difference, 
+			                                              	  xEuler ) );
```

Comment:
You should just do the math instead of making this a lerp between 1 and 0.
I think that the whole line is equivalent to speedMod = 1 - (xEuler - _rotationBounds.Difference + _rotationBoundsWidth) / _rotationBoundsWidth

work:
inverseLerp(a, b, x ) = (x - a) / (b - a)

inverseLerp(D - W, D, x) = 
(x - (D - W)) / (D - (D - W) = ( x - D + W ) / W